### PR TITLE
API/DEPR: 'periods' argument instead of 'n' for PeriodIndex.shift()

### LINKED ISSUE #22458: Inconsistent keyword argument name for the 'periods' in pandas.Index.shift for a DatetimeIndex
The actual implementation of ``pandas.Index.shift`` for a datetime-like index takes a differently named parameter for the number of shifts than the base method.

It should be named ``periods``, which would also be consistent with the ``pandas.DataFrame.shift`` or ``pandas.Series.shift`` methods. But for a DatetimeIndex it's ``n``.

See:
https://github.com/pandas-dev/pandas/blob/v0.23.4/pandas/core/indexes/base.py#L2586
vs.
https://github.com/pandas-dev/pandas/blob/v0.23.4/pandas/core/indexes/datetimelike.py#L1021

With a DatetimeIndex, following the documentation leads to a
```
TypeError: shift() got an unexpected keyword argument 'periods'
```

If this is somehow intended, then it should be stated so in the docstring for ``pandas.Index.shift``.

pandas version is 0.23.4

### PR DESCRIPTION
**Pull request by arminv**
- [x] closes #22458 
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

In order to be consistent with `Index.shift` & `Series.shift` & `DatetimeIndex.shift`, `n` argument was deprecated in favor of `periods`.

```
In [2]: idx = pd.PeriodIndex(freq='A', start='1/1/2001', end='12/1/2009')

In [3]: idx
Out[3]: 
PeriodIndex(['2001', '2002', '2003', '2004', '2005', '2006', '2007', '2008',
             '2009'],
            dtype='period[A-DEC]', freq='A-DEC')

In [4]: idx.shift(1)
Out[4]: 
PeriodIndex(['2002', '2003', '2004', '2005', '2006', '2007', '2008', '2009',
             '2010'],
            dtype='period[A-DEC]', freq='A-DEC')

In [5]: idx.shift(periods=1)
Out[5]: 
PeriodIndex(['2002', '2003', '2004', '2005', '2006', '2007', '2008', '2009',
             '2010'],
            dtype='period[A-DEC]', freq='A-DEC')

In [6]: idx.shift(n=1)
//anaconda/envs/pandas_dev/bin/ipython:1: FutureWarning: the 'n' keyword is deprecated, use 'periods' instead
  #!//anaconda/envs/pandas_dev/bin/python
Out[6]: 
PeriodIndex(['2002', '2003', '2004', '2005', '2006', '2007', '2008', '2009',
             '2010'],
            dtype='period[A-DEC]', freq='A-DEC')
```

### CODE REVIEW

**Review by TomAugspurger — COMMENTED** [review]
LGTM.

I didn't closely follow the issue and other discussion, but I just want to double-check that we really *want* to do this. DatetimeIndex / PeriodIndex.shift are fundamentally different from Series.shift, (shifting values vs. position). Still, I think changing the argument name is still for the best.

cc @jorisvandenbossche

**Comment by jreback** [comment]
this is ok, we have an inconsistency in named args now.

File: pandas/core/arrays/period.py [hunk 1]
```diff
@@ -319,20 +319,33 @@ def _add_delta(self, other):
...
 
         Returns
         -------
-        shifted : Period Array/Index
+        pandas.PeriodIndex
+            Shifted index.
+
+        See Also
+        --------
+        Index.shift : Shift values of Index.
```
[the remark this answers is not in the document]
Comment (jreback) [reply pandas/core/arrays/period.py]: can you update this @arminv

File: doc/source/whatsnew/v0.24.0.txt [hunk 2]
```diff
@@ -573,7 +573,7 @@ Deprecations
 - :meth:`Series.str.cat` has deprecated using arbitrary list-likes *within* list-likes. A list-like container may still contain
   many ``Series``, ``Index`` or 1-dimensional ``np.ndarray``, or alternatively, only scalar values. (:issue:`21950`)
 - :meth:`FrozenNDArray.searchsorted` has deprecated the ``v`` parameter in favor of ``value`` (:issue:`14645`)
-- :func:`DatetimeIndex.shift` now accepts ``periods`` argument instead of ``n`` for consistency with :func:`Index.shift` and :func:`Series.shift`. Using ``n`` throws a deprecation warning (:issue:`22458`)
+- :func:`DatetimeIndex.shift` and :func:`PeriodIndex.shift` now accept ``periods`` argument instead of ``n`` for consistency with :func:`Index.shift` and :func:`Series.shift`. Using ``n`` throws a deprecation warning (:issue:`22458`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.24.0.txt]: can you add this issue as well here (this PR number is fine)

**Review by jreback — CHANGES_REQUESTED** [review]
ping on green.

**Review by TomAugspurger — APPROVED** [review]
Sorry about the merge conflict. This looks good.

**Comment by arminv** [comment]
@jreback all green.
@TomAugspurger thanks for reviewing.

**Comment by TomAugspurger** [comment]
Thanks!

**Pull request merged**